lib/src/host/preflight.rs

```diff
@@ -340,7 +340,9 @@ fn proofs_to_tries(
 
         // assure that slots can be deleted from the storage trie
         for storage_proof in &fini_proofs.storage_proof {
-            add_orphaned_leafs(storage_proof.key, &storage_proof.proof, &mut storage_nodes)?;
+            use ethers_core::abi::ethabi::ethereum_types::BigEndianHash;
+            let storage_proof_key = ethers_core::types::H256::from_uint(&storage_proof.key);
+            add_orphaned_leafs(storage_proof_key, &storage_proof.proof, &mut storage_nodes)?;
```

Comment:
This also works, and might be a bit nicer. Not sure, whatever you prefer...

```suggestion
            let key = from_ethers_u256(storage_proof.key);
            add_orphaned_leafs(
                key.to_be_bytes::<32>(),
                &storage_proof.proof,
                &mut storage_nodes,
            )?;
```